### ISSUE TITLE
question about pic1 in the paper

### ISSUE THREAD

**Issue by xhc19930714** [AIDC-AI/Marco-o1#3]
![2024-11-23_010752](https://github.com/user-attachments/assets/e5ec5485-f766-4e54-ba72-25f9de7b2caa)
as shown in pic1, why did not count all letters in strawberry? Just count to the second-to-last letter and it's over

**Comment by Sniper970119** [comment]
Ineresting discovery! 
We reviewed the original output, and it seems that the model indeed does not compare the last 'y' in this issue (meaning it's not due to a typo or intentional omission). We found that in the search paths of MCTS, most of the 'y's could be ignored by the model, but there are still some paths where the 'y' is compared.
We did not intentionally guide this behavior, so we tend to believe that the model was so certain about the last token that it did not output it (perhaps similar to how we might omit certain steps in a math problem when we are confident about them). However, this is still our conjecture, and we welcome everyone to join in the discussion on this issue.
Thank you for your attention. We will add this intersting findings in Figure 1 caption.


The paths mostly don't compare the last letter 'y'.
![image](https://github.com/user-attachments/assets/0a27e092-e0eb-450a-9188-f35ff2ff9c0d)
![image](https://github.com/user-attachments/assets/4921c243-4fcb-489e-95b4-a004e03d24d1)

Some paths do thoroughly compare all the letters of this word.
![image](https://github.com/user-attachments/assets/509586de-13e0-4130-9615-201149e6993b)
